apps/meteor/app/livechat/server/lib/contacts/disableContact.ts

```diff
@@ -0,0 +1,16 @@
+import { LivechatContacts } from '@rocket.chat/models';
+
+import { settings } from '../../../../settings/server';
+
+export async function disableContactById(contactId: string): Promise<void> {
+	if (!settings.get('Omnichannel_enable_contact_removal')) {
+		throw new Error('error-contact-removal-disabled');
+	}
+
+	const contact = await LivechatContacts.findOneById(contactId);
```

Comment:
you don't need to fetch the whole contact object just to validate it exists.. instead you can used the return of the `update` you're doing later (when you call `LivechatContacts.disableByContactId`) for that.. please check: https://mongodb.github.io/node-mongodb-native/6.18/interfaces/UpdateResult.html

you can use `matchedCount` for that.